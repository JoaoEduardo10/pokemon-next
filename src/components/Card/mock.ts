import { ApiPokemonProps } from '../../types';

export const mockPorkemon: ApiPokemonProps = {
	id: 1,
	name: 'picachu',
	url: 'https://pokemon'
};