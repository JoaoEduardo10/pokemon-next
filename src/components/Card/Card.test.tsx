import { render, screen } from '@testing-library/react';
import { Card } from '.';
import { mockPorkemon } from './mock';

describe('<Card />', () => {
	it('should render a Card', () => {
		const { container } = render(<Card pokemon={mockPorkemon} />);

		expect(screen.getByRole('img', { name: 'picachu' })).toBeInTheDocument();
		expect(screen.getByLabelText('numero do pokemon')).toBeInTheDocument();
		expect(screen.getByRole('heading', { name: 'picachu' })).toBeInTheDocument();
		expect(screen.getByRole('link', { name: 'Detalhes' })).toBeInTheDocument();
		expect(screen.getByRole('link', { name: 'Detalhes' })).toHaveAttribute('href', `/pokemon/${mockPorkemon.id}`);

		expect(container).toMatchSnapshot();
	});
});