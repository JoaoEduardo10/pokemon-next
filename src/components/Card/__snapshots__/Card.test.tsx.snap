// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render a Card 1`] = `
<div>
  <li
    class="card"
  >
    <img
      alt="picachu"
      data-nimg="1"
      decoding="async"
      height="180"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2FPokeAPI%2Fsprites%2Fmaster%2Fsprites%2Fpokemon%2F1.png&w=384&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2FPokeAPI%2Fsprites%2Fmaster%2Fsprites%2Fpokemon%2F1.png&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2FPokeAPI%2Fsprites%2Fmaster%2Fsprites%2Fpokemon%2F1.png&w=384&q=75 2x"
      style="color: transparent;"
      width="180"
    />
    <p
      aria-label="numero do pokemon"
      class="id"
    >
      #
      1
    </p>
    <h3
      class="title"
    >
      picachu
    </h3>
    <a
      class="btn"
      href="/pokemon/1"
    >
      Detalhes
    </a>
  </li>
</div>
`;
